components/viaduct/android/src/main/java/mozilla/appservices/httpconfig/LibViaduct.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.httpconfig
+
+import android.util.Log
+import com.sun.jna.Library
+import com.sun.jna.Callback
+import com.sun.jna.Native
+import java.lang.reflect.Proxy
+import mozilla.appservices.support.RustBuffer
+
+@Suppress("FunctionNaming", "TooGenericExceptionThrown")
+internal interface LibViaduct : Library {
+    companion object {
+        private val JNA_LIBRARY_NAME = {
+            val libname = System.getProperty("mozilla.appservices.viaduct_lib_name")
+            if (libname != null) {
+                Log.i("AppServices", "Using viaduct_lib_name: " + libname)
+                libname
+            } else {
+                "viaduct"
+            }
+        }()
+
+        internal var INSTANCE: LibViaduct = try {
+            val lib = Native.loadLibrary(JNA_LIBRARY_NAME, LibViaduct::class.java) as LibViaduct
+            if (JNA_LIBRARY_NAME == "viaduct") {
+                // TODO Enable logging if we aren't in a megazord.
+            } else {
+                // We're in a megazord. If the build is properly configured,
+                // then this is effectively a no-op since . However, there are
+                // a lot of variables and things cargo doesn't exactly
+                // guarantee that we're relying on, so in the future it
+                // is possible that somehow a build ships that contains the
+                // rust backend by accident. Calling this means we'll still
+                // make our requests through OpenSSL.
```

Comment:
The comment suggests that we'll make our requests through OpenSSL, but the code suggests it will force us to make requests over the FFI via concept-fetch; which is correct?

Reply:
Whoops, typo, that should say Necko.